src/server/utilities/liteATIClickTracking/index.test.ts

```diff
@@ -0,0 +1,171 @@
+import { LITE_ATI_TRACKING } from '#app/hooks/useClickTrackerHandler';
+import liteATIClickTracking from '.';
+
+const dispatchClick = (targetElement: HTMLElement) => {
+  document.body.appendChild(targetElement);
+  const event = new MouseEvent('click', {
+    view: window,
+    bubbles: true,
+    cancelable: true,
+  });
+  targetElement.dispatchEvent(event);
+};
+
+describe('Click tracking script', () => {
+  const originalWindowLocation = window.location;
+
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date('2024-11-13T16:30:02.000Z'));
+    let mockCookie = '';
+    Object.defineProperty(document, 'cookie', {
+      get() {
+        return mockCookie;
+      },
+      set(cookieValue) {
+        mockCookie = cookieValue;
+      },
+    });
+    Object.defineProperty(window, 'location', {
+      value: {
+        assign: jest.fn(),
+      },
+    });
+    liteATIClickTracking();
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    document.cookie = '';
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+    window.sendBeaconLite = jest.fn();
+    window.dispatchEvent(new Event('load'));
+  });
+
+  afterAll(() => {
+    Object.defineProperty(window, 'location', {
+      value: {
+        ...originalWindowLocation,
+      },
+    });
+  });
+
+  it('Does not call sendBeacon if the event has no data-ati-tracking parameter, but still redirects', () => {
+    const anchorElement = document.createElement('a');
+    anchorElement.href = '/gahuza';
+
+    dispatchClick(anchorElement);
+
+    expect(window.sendBeaconLite).toHaveBeenCalledTimes(0);
+    const nextPageUrl = (window.location.assign as jest.Mock).mock.calls[0][0];
+    expect(nextPageUrl).toContain('/gahuza');
+  });
+
+  it('Does not add userId cookie if crypto is invalid, but stil calls sendBeacon', () => {
+    const anchorElement = document.createElement('a');
+    anchorElement.setAttribute(
+      LITE_ATI_TRACKING,
+      'https://logws1363.ati-host.net/?',
+    );
+
+    // @ts-expect-error Some browsers may not have crypto.
+    // eslint-disable-next-line no-global-assign
+    crypto = undefined;
+    dispatchClick(anchorElement);
+
+    const callParam = (window.sendBeaconLite as jest.Mock).mock.calls[0][0];
+    const parsedATIParams = Object.fromEntries(new URLSearchParams(callParam));
+    expect(parsedATIParams.idclient).toBe(undefined);
```

Comment:
Or use this expectation?
```suggestion
    expect(parsedATIParams.idclient).toBeUndefined();
```